AirCasting/Services/RemoveOldMeasurementsService.swift

```diff
@@ -21,20 +20,41 @@ final class RemoveOldMeasurementsService {
             try removeMeasurementsSurplus(context: context, streams: session.allStreams)
         } else {
             let session = try context.existingExternalSession(uuid: sessionUUID)
-            try removeMeasurementsSurplus(context: context, streams: session.allStreams)
+            try removeMeasurementsSurplus(context: context, streams: session.allStreams, provider: session.provider)
         }
     }
     
-    private func removeMeasurementsSurplus(context: NSManagedObjectContext, streams: [MeasurementStreamEntity]) throws {
+    // PurpleAir — measurements from there are not consistent, so we need to do more 'manual work' 🔧
+    // OpenAQ — measurements from there are not consistent, so we need to do more 'manual work' 🔧
+    // 'manual work' - taking the newest measurement and by having its time we are going back exactly 24 hours
+    // AirBeam - measurement every 1 min (60 times/1h): 1440 measurements (per 24 hours)
```

Comment:
what happened here is a call for refactor of this class:
1. Create a protocol for this class:
2. Create multiple implementations - one for each provider
3. Register them in `Resolver` and inject based on the argument which would be a `SessionUUID`
4. Inject wherever it's needed

Reply:
I am not sure about 4th one. I will try to do it and if you could please check then